File: app/src/main/java/com/feriantes4dawin/feriavirtualmovil/data/repos/UsuarioRepository.java
package com.feriantes4dawin.feriavirtualmovil.data.repos;

import androidx.lifecycle.LiveData;

import com.feriantes4dawin.feriavirtualmovil.data.Result;
import com.feriantes4dawin.feriavirtualmovil.data.models.Usuario;

import javax.inject.Singleton;

import dagger.Component;
import dagger.Module;
import dagger.Provides;


public interface UsuarioRepository {

    UsuarioRepository getInstance();

    Result<LiveData<Usuario>> loginUsuario(String email, String contrasena);

    Result<LiveData<Usuario>> getInfoUsuario(Usuario u);

    boolean updateUsuario(int ntelefono,String ndireccion,String npasswd);
}

File: app/src/main/java/com/feriantes4dawin/feriavirtual/data/repos/UsuarioRepository.kt
package com.feriantes4dawin.feriavirtual.data.repos

import androidx.lifecycle.LiveData
import com.feriantes4dawin.feriavirtual.data.db.UsuarioDAO
import com.feriantes4dawin.feriavirtual.data.model.Usuario
import com.feriantes4dawin.feriavirtual.data.network.responses.UsuarioResponse

interface UsuarioRepository {
    suspend fun loginUsuario(email:String,contrasena:String): Boolean
    suspend fun getInfoUsuario(u: Usuario) : LiveData<Usuario>
}